### PR TITLE
Add trace to API logs

### PR DESCRIPTION
**Pull request by michaelkedar**
Adds the trace fields to most of the logs produced by the API, so it's easier correlate the logs to their requests.

Relevant links:
https://cloud.google.com/trace/docs/setup#force-trace
https://cloud.google.com/logging/docs/agent/logging/configuration#special-fields

### CODE REVIEW

File: gcp/api/server.py [hunk 1]
```diff
@@ -102,6 +102,31 @@ def wrapper(*args, **kwargs):
...
 
 
+def trace_log_fields(context: grpc.ServicerContext) -> dict:
+  """Makes the json_field needed to associate a log with the request trace."""
+  fields = {}
+  trace_context = dict(
+      context.invocation_metadata()).get('x-cloud-trace-context')
+  if trace_context is None:
+    return fields
+
```
Comment (another-rex) [inline gcp/api/server.py:112]:
```suggestion

# Trace context header example: "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
```

**Review by another-rex — APPROVED** [review]
Nice! This will make looking at the logs so much easier!

**Review by oliverchang — APPROVED** [review]
LGTM, but is there a way we can  generalize this by doing this in https://github.com/google/osv.dev/blob/master/osv/logs.py instead? 

Perhaps we can set some kind of thread local storage to propagate the context to the logs.py `filter` function? 

Can be a follow up PR.

**Comment by michaelkedar** [comment]
> LGTM, but is there a way we can generalize this by doing this in https://github.com/google/osv.dev/blob/master/osv/logs.py instead?
> 
> Perhaps we can set some kind of thread local storage to propagate the context to the logs.py `filter` function?
> 
> Can be a follow up PR.

Ack. Will look into this in another PR.
Though, I'm not sure this would belong in logs.py - the API would be the only thing to use it.

**Pull request merged**